src/json.c

```diff
@@ -768,6 +775,9 @@ lyjson_ctx_next(struct lyjson_ctx *jsonctx, enum LYJSON_PARSER_STATUS *status)
             ret = lyjson_value(jsonctx);
         }
     } else if (((prev == LYJSON_OBJECT) && (*jsonctx->in->current == '}')) || ((prev == LYJSON_ARRAY) && (*jsonctx->in->current == ']'))) {
+        if (*jsonctx->in->current == '}') {
+            jsonctx->depth = jsonctx->depth ? jsonctx->depth - 1 : 0;
```

Comment:
The zero-check `assert()` should also be here, after checking that it really must not occur and of it can, handle it accordingly.